tap_pardot/client.py

```diff
@@ -85,7 +88,13 @@ def _make_request(self, method, url, params=None):
             url,
             params,
         )
-        response = requests.request(
+
+        request_session = requests.Session()
```

Comment:
try and make a top-level session instead of a session per request. You can create one in the `init` function and simply re-use it. It is quite a boost to performance, because it can use the keep-alive stuff and cache a lot of things for you. It is not a dramatic performance things, but it requires little to no work and is a great thing to just do for that one time when performance does matter.

It also allows you to do the retry stuff once instead of on every request.

Reply:
I understand it, I don't know why I have missed that :p probably I was distracted by many different and new things for me